emu/src/cpu/arm7tdmi.rs

```diff
@@ -1019,6 +1019,24 @@ impl Arm7tdmi {
         Some(SIZE_OF_THUMB_INSTRUCTION)
     }
 
+    fn load_address(&mut self, op_code: ThumbModeOpcode) -> Option<u32> {
+        let sp = op_code.get_bit(11);
+        let rd: usize = op_code.get_bits(8..=10).try_into().unwrap();
+        let word8: u32 = op_code.get_bits(0..=7).into();
+        let word8 = word8 << 2;
+
+        let v = if sp {
+            let stack_pointer = self.registers.register_at(REG_SP);
+            stack_pointer + word8
+        } else {
+            let pc = self.registers.program_counter() as u32;
```

Comment:
If we use `pc` we have to add `4`:

> Where the PC is used as the source register (SP = 0), bit 1 of the PC is always read as 0. The value of the PC will be 4 bytes greater than the address of the instruction before bit 1 is forced to 0.

We should also set the bit `1` to `0` since this is THUMB mode and it might happen that the bit 1 is `1` (since thumb instructions are 16 bits)

Reply:
Done